### PR TITLE
Split up dex-data into more files

### PR DESCRIPTION
**Pull request by Zarel**
This commit introduces:
- `dex-abilities.ts`
- `dex-conditions.ts`
- `dex-formats.ts`
- `dex-items.ts`
- `dex-moves.ts`
- `dex-species.ts`

These files centralize definitions from `dex-data` and `global-types`.

### CODE REVIEW

**Comment by Zarel** [comment]
Next steps:

- Move definitions from `Move` to `DataMove` so they can be merged into a single `Move` type/class (and similarly for the others)
- Start work on new Dex API (`Dex.moves.get()`, `Dex.moves.all()`)

File: sim/dex.ts [hunk 1]
```diff
@@ -1596,3 +1533,13 @@ dexes['base'] = new ModdedDex(undefined, true);
...
+export namespace Dex {
+	export type Species = import('./dex-species').Species;
+	export type Item = import('./dex-items').Item;
+	export type Move = import('./dex-moves').Move;
+	export type Ability = import('./dex-abilities').Ability;
+
+	export type HitEffect = import('./dex-moves').HitEffect;
+	export type SecondaryEffect = import('./dex-moves').SecondaryEffect;
+	export type RuleTable = import('./dex-formats').RuleTable;
+}
```
Comment (Zarel) [inline sim/dex.ts:1545]: Am very annoyed I have to do this but I don't see any other way. TypeScript being TypeScript... :/

File: data/scripts.ts [hunk 2]
```diff
@@ -1048,7 +1050,7 @@ export const Scripts: BattleScriptsData = {
 		if (!moveData.secondaries) return;
 		for (const target of targets) {
 			if (target === false) continue;
-			const secondaries: SecondaryEffect[] =
+			const secondaries: Dex.SecondaryEffect[] =
```
Comment (KrisXV) [inline data/scripts.ts:1053]: ? Shouldn’t this be `this.dex`?

Comment (Zarel) [reply data/scripts.ts:1053]: No, it's a type.

Comment (Zarel) [reply data/scripts.ts:1053]: It does not even seem possible to define a type in `this.dex` :(

Comment (KrisXV) [reply data/scripts.ts:1053]: oh im dumb

also, rip

**Pull request merged**